### PR TITLE
#152798773 Integrate payment using paystack

### PR DESCRIPTION
**Pull request by paradisekelechi**
#### What does this PR do?
Integrates payment using the paystack
#### Description of Task to be completed?
- Define structure for the payment
- Integrate paystack payment for client users
- Add settings for the admin to add the public and secret keys where necessary
#### How should this be manually tested?
Load the application, checkout and pay accordingly
#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
#152798773
#### Screenshots (if appropriate)
image

image

#### Questions:
Not applicable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
+[![bitHound Overall Score](https://www.bithound.io/github/reactioncommerce/reaction/badges/score.svg)](https://www.bithound.io/github/reactioncommerce/reaction) [![bitHound Dev Dependencies](https://www.bithound.io/github/reactioncommerce/reaction/badges/devDependencies.svg)](https://www.bithound.io/github/reactioncommerce/reaction/9a858eb459d7260d5ae59124c2b364bc791a3e70/dependencies/npm) [![bitHound Code](https://www.bithound.io/github/reactioncommerce/reaction/badges/code.svg)](https://www.bithound.io/github/reactioncommerce/reaction) [![Circle CI](https://circleci.com/gh/andela/mont-blanc-rc.svg?style=svg)](https://circleci.com/gh/andela/mont-blanc-rc) [![Gitter](https://badges.gitter.im/JoinChat.svg)](https://gitter.im/reactioncommerce/reaction?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
```
Comment (shemogumbe) [inline README.md]: These are change son badges, how relevant are they on add-payment task, these changes are not necessary for this task

File: .meteor/versions [hunk 2]
```diff
@@ -148,6 +149,8 @@ service-configuration@1.0.11
 session@1.1.7
 sha@1.0.9
 shell-server@0.2.4
+simple:json-routes@2.1.0
```
Comment (shemogumbe) [inline .meteor/versions]: This line and the one below it are not relevannt for the payment task

File: .meteor/versions [hunk 3]
```diff
@@ -120,6 +120,7 @@ mongo-id@1.0.6
 mongo-livedata@1.0.12
 mrt:later@1.6.1
 natestrauser:select2@4.0.3
+nimble:restivus@0.8.12
```
Comment (shemogumbe) [inline .meteor/versions]: This is equally valuable for making end points restful, however, it has no relationship with payment method and thus is an unnecessary file change

File: .meteor/packages [hunk 4]
```diff
@@ -94,3 +94,5 @@ johanbrook:publication-collector
 # meteorhacks:sikka                         # additional ddp, login security
 
 # Custom Packages
+simple:rest
```
Comment (shemogumbe) [inline .meteor/packages]: These are not necessary for this task

File: .eslintrc [hunk 5]
```diff
@@ -11,33 +11,34 @@
...
-        "objectLiteralDuplicateProperties": false,
-        "objectLiteralShorthandMethods": true,
-        "objectLiteralShorthandProperties": true,
-        "spread": true,
-        "superInFunctions": true,
-        "templateStrings": true,
-        "jsx": true
-      },
-     "allowImportExportEverywhere": true
+    "ecmaVersion": 6,
```
Comment (shemogumbe) [inline .eslintrc]: Are all these changes necessary to have for the payment feature to work, I guesse they should not be here then

File: imports/plugins/included/payments-paystack/lib/collections/schemas/paystack.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    }
+  }
+]);
+
+registerSchema("PaystackPackageConfig", PaystackPackageConfig);
+
+export const PaystackPayment = new SimpleSchema({
+  payerName: {
+    type: String,
+    label: "Cardholder name"
```
[the remark this answers is not in the document]
Comment (shemogumbe) [reply imports/plugins/included/payments-paystack/lib/collections/schemas/paystack.js]: On style guides, strings within objects should be single quotes

File: .eslintrc [hunk 7]
```diff
@@ -15,27 +15,11 @@
     "PaystackPop": true
   },
   "parserOptions": {
-    "ecmaVersion": 6,
+    "ecmaVersion": 8,
```
Comment (shemogumbe) [inline .eslintrc]: Is this change necessary for this PR

File: .meteor/versions (hunk 3 above)
Comment (paradisekelechi) [reply .meteor/versions]: Oh, that is true. I am working on it

File: .eslintrc (hunk 5 above)
Comment (paradisekelechi) [reply .eslintrc]: Yes. This change is particularly necessary. I added the PaystackPop and Alerts as a global variable.
Consequently, a restructuring of the indentation

Comment (shemogumbe) [reply .eslintrc]: The line  "PaystackPop": true is the only one that is worth having, right, what of the ecmaversion, alerts etc?

**Review by shemogumbe — APPROVED** [review]
